packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -816,6 +816,10 @@ export function getValidatorApi({
     },
 
     async registerValidator(registrations) {
+      if (!chain.executionBuilder) {
```

Comment:
This API will now throw an error if a validator client tries to register validators and the beacon node does not have `--builder` flag set. This is quite important in my opinion as current behavior is that this API just succeeds without actually registering validators with the builder. Without this check, the earliest we would detect a missing builder configuration on the beacon node would be when trying to produce a blinded block.